common/src/parse.cpp

```diff
@@ -156,10 +156,8 @@ pcl::console::parse_file_extension_argument (int argc, const char * const * argv
   for (int i = 1; i < argc; ++i)
   {
     std::string fname = std::string (argv[i]);
-    for (size_t j = 0; j < extension.size (); ++j)
+    for (auto ext : extension)
```

Comment:
Just as note: This is not as reference, because value will be modified (and modified value should not be stored in `extension`)